### ISSUE TITLE
shouldn't use profile role and name interchangeably when constructing scaleset name

### ISSUE THREAD

**Issue by jim-minter** [openshift/openshift-azure#894]
@openshift/sig-azure 

I think we have a typing issue in at least pkg/cluster and pkg/arm where it comes to how we name scalesets.  Sometimes we do `fmt.Sprintf("ss-%s_%d", profile.Name, ...)`; sometimes we do `"ss-"+string(role)`.

Alarmingly I think *both* of the options above could potentially be wrong: the first because I suspect we don't validate the profile name strongly enough to convert it direct to a scaleset name (e.g. we allow upper and lower cases - is that a problem?  Is Azure's case sensitivity in this space the same as ours?)

The second is trivially broken whenever the compute agent pool profile isn't called "compute", and I don't think we actually enforce this anywhere any more? (or do we?)

@asalkeld have you got time to take a look at this?  I think it's going to need a thoughtful approach to avoid getting this wrong again.

**Comment by asalkeld** [comment]
https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions
It doesn't specifically talk about scalesets, but i'd assume the rules for compute
-Case insensitive
-Alphanumeric and hyphen

As far as validation:
- We check the roles here: https://github.com/openshift/openshift-azure/blob/master/pkg/api/validate.go#L344
- Infra name and role have to be the same: https://github.com/openshift/openshift-azure/blob/master/pkg/api/validate.go#L391
- we don't seem to check master's name (I can fix that)
- we check that compute != master or infra
- we check compute with this regex https://github.com/openshift/openshift-azure/blob/master/pkg/api/validate.go#L27 (which seems ok)

So basically I think we should write a wrapper function that does the first option:
```
func getScalesetName(cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole) scalesetName {
	for _, app := range cs.Properties.AgentPoolProfiles {
		if app.Role == role {
			return scalesetName("ss-" + app.Name)
		}
	}
	panic("invalid role")
}
```

**Comment by jim-minter** [comment]
#916 doesn't solve this issue (I didn't describe it clearly) re-opening.

**Comment by jim-minter** [comment]
was fixed as part of the update rework
